tests/test_lorentz.py

```diff
@@ -283,7 +294,167 @@ def cos_theta_hat_2_canonical_3(M, m1, m2, m3, sigma1, sigma2, sigma3):
+    assert round(masssq(p3), 5) == round(m3sq, 5)
+    assert round(masssq(p1 + p2), 5) == round(m12sq, 5)
+    assert round(masssq(p2 + p3), 5) == round(m23sq, 5)
+    assert round(masssq(p3 + p1), 5) == round(m31sq, 5)
+
+    # Lorentz transformation
+    momenta = {i: p for i, p in zip([1, 2, 3], [p1, p2, p3])}
+    tree1 = Topology(root=0, decay_topology=((2, 3), 1))
+    tree2 = Topology(root=0, decay_topology=((3, 1), 2))
+    tree3 = Topology(root=0, decay_topology=((1, 2), 3))
+    # momenta = tc.topologies[0].to_rest_frame(momenta)
+    momenta = tree1.align_with_daughter(momenta, (2, 3))
+    phi_rf = chain_vars["Kpi"]["phi_Kst"]
+    theta_rf = chain_vars["Kpi"]["theta_Kst"]
+    psi_rf = chain_vars["Kpi"]["phi_K"]
+
+    rotation = LorentzTrafo(0, 0, 0, phi_rf, theta_rf, psi_rf)
+    # direct outside rotations are not really supported, but possible via direct matrix multiplication or via the root node of a tree
+    momenta = tree1.to_rest_frame(momenta)
+    momenta_23_rotated = tree1.root.transform(rotation, momenta)
+    assert np.allclose(
+        np.cos(tree1.helicity_angles(momenta_23_rotated)[((2, 3), 1)].theta_rf),
+        np.cos(chain_vars["Kpi"]["theta_Kst"]),
+    )
+    assert np.allclose(
+        np.cos(tree1.helicity_angles(momenta_23_rotated)[((2, 3), 1)].psi_rf),
+        np.cos(chain_vars["Kpi"]["phi_Kst"]),
+    )
+    assert np.allclose(
+        np.cos(tree1.helicity_angles(momenta_23_rotated)[(2, 3)].psi_rf),
+        np.cos(chain_vars["Kpi"]["phi_K"]),
+    )
+    assert np.allclose(
+        np.cos(tree1.helicity_angles(momenta_23_rotated)[(2, 3)].theta_rf),
+        np.cos(chain_vars["Kpi"]["theta_K"]),
+        1e-4,
+    )
+
+    assert np.allclose(
+        np.cos(tree2.helicity_angles(momenta_23_rotated)[((3, 1), 2)].theta_rf),
+        np.cos(chain_vars["pip"]["theta_D"]),
+    )
+    assert np.allclose(
+        np.cos(tree2.helicity_angles(momenta_23_rotated)[((3, 1), 2)].psi_rf),
+        np.cos(chain_vars["pip"]["phi_D"]),
+    )
+
+    assert np.allclose(
+        np.cos(tree2.helicity_angles(momenta_23_rotated)[(3, 1)].psi_rf),
+        np.cos(chain_vars["pip"]["phi_pi"]),
+    )
+    assert np.allclose(
+        np.cos(tree2.helicity_angles(momenta_23_rotated)[(3, 1)].theta_rf),
+        np.cos(chain_vars["pip"]["theta_pi"]),
+    )
+
+    assert np.allclose(
+        np.cos(tree3.helicity_angles(momenta_23_rotated)[((1, 2), 3)].theta_rf),
+        np.cos(chain_vars["pK"]["theta_L"]),
+    )
+    assert np.allclose(
+        np.cos(tree3.helicity_angles(momenta_23_rotated)[((1, 2), 3)].psi_rf),
+        np.cos(chain_vars["pK"]["phi_L"]),
+    )
+    assert np.allclose(
+        np.cos(tree3.helicity_angles(momenta_23_rotated)[(1, 2)].psi_rf),
+        np.cos(chain_vars["pK"]["Lphi_p"]),
+        1e-4,
+    )
+    assert np.allclose(
+        np.cos(tree3.helicity_angles(momenta_23_rotated)[(1, 2)].theta_rf),
+        np.cos(chain_vars["pK"]["Ltheta_p"]),
+        1e-4,
+    )
+
```

Comment:
This is the actual new test. 
Sorry i smuggeled some cleanups in theis PR aswell. 
I used the direct definition of Topologies here, so we would not be bothered by translating dpd definitions into what decayangle does normally